src/providers/chatwoot.ts

```diff
@@ -0,0 +1,55 @@
+import { State } from 'types'
+
+const domain = 'https://app.chatwoot.com'
+
+declare global {
+  interface Window {
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    chatwootSettings: any
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    $chatwoot: any
+    chatwootSDK: {
+      run: (options: { websiteToken: string; baseUrl: string }) => void
+    }
+  }
+}
+
+/* eslint-disable */
+const loadScript = (onload: () => void) => {
+  if (window.$chatwoot) return
+  ;(function(d, t) {
+    var script: HTMLScriptElement = d.createElement('script')
+    var fisrtScript = d.getElementsByTagName('script')[0]
+    script.src = domain + '/packs/js/sdk.js'
+    fisrtScript.parentNode?.insertBefore(script, fisrtScript)
+    script.onload = onload
+  })(document)
+}
+/* eslint:enable */
+
+const load = ({
+  providerKey,
+  setState
+}: {
+  providerKey: string
+  setState: (state: State) => void
+}) => {
+  loadScript(function() {
+    setTimeout(() => setState('complete'), 1000)
+    window.chatwootSDK.run({
+      websiteToken: providerKey,
+      baseUrl: domain
```

Comment:
I couldn't find reference to this in the docs so please link me if it's there, but should this be the `domain` (as in, the chatwoot domain) or should it be a user provided value for their site?

Reply:
It could be both. Chatwoot has a SaaS option with `https://app.chatwoot.com` domain. Additionally users are aple to self host Chatwoot with custom domain.  
What would be the proper way to read from user settings?

Reply:
We did a similar thing with `messenger` where we get the `appId` and `locale` from `LiveChatProvider`
- https://github.com/calibreapp/react-live-chat-loader/blob/main/src/providers/messenger.ts#L41
- https://github.com/calibreapp/react-live-chat-loader/blob/main/src/hooks/useChat.ts#L30

I think we could actually just spread these props so in the future if you need an extra optional prop it'll flow through from `LiveChatLoaderProvider` to here and you can use it as required.

Reply:
I will ads support for [Chatwoot SDK settings](https://www.chatwoot.com/docs/product/channels/live-chat/sdk/setup#sdk-settings).

```js
window.chatwootSettings = {
  hideMessageBubble: false,
  position: 'left', // This can be left or right
  locale: 'en', // Language to be set
  type: 'standard', // [standard, expanded_bubble]
};
```

After that it can be merged, after #75 fixes the build I guess